common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java

```diff
@@ -163,14 +163,11 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         if (e.state() == IdleState.ALL_IDLE && isActuallyOverdue) {
           if (responseHandler.hasOutstandingRequests()) {
             String address = getRemoteAddress(ctx.channel());
-            logger.error("Connection to {} has been quiet for {} ms while there are outstanding " +
-              "requests. Assuming connection is dead; please adjust" +
-              " spark.{}.io.connectionTimeout if this is wrong.",
-              address, requestTimeoutNs / 1000 / 1000, transportContext.getConf().getModuleName());
-            client.timeOut();
-            ctx.close();
```

Comment:
if `closeIdleConnection` is `true`, idle connection still be closed. If false, the client will wait forever. I have followup change to allow user to config `closeIdleConnection` and `requestTimeout`.

Reply:
The closeIdleConnection flag determines whether to close the connection, regardless of whether there are any pending requests. When this flag is enabled, the connection is closed even if there are no outstanding requests. However, if there are pending requests, the client adheres to the requestTimeoutNs duration, which is a sensible approach because waiting indefinitely is not recommended. Modifying this behavior would result in a change to long-established semantics.